eZ/Publish/SPI/Persistence/Content/CreateStruct.php

```diff
@@ -72,4 +72,11 @@ class CreateStruct extends ValueObject
      * @var int
      */
     public $modified;
+
+    /**
+     * Is hidden flag.
+     *
+     * @var bool
+     */
+    public $isHidden;
```

Comment:
Any reason this can't be in the API?  Would help for those of use that want to create content that is hidden by default or change the default state to hidden in an event listener.

Currently I do this but have have to change the visibility after it is published.  I would love to avoid having it visible for a split second and having the cache updated twice.

Reply:
No strong opinion on this one, @ViniTou, @alongosz what is your POV?

Reply:
@konradoboza With more thought this will not improve my issue.  It would allow to set the visibility of the initially created draft but that would not be sufficient as I use field values to determine if it should be hidden or not in the handler.  hideContent/revealContent in the publish event looks like the best way to go still.

Reply:
> Any reason this can't be in the API? Would help for those of use that want to create content that is hidden by default or change the default state to hidden in an publish event listener (BeforeCreateContentEvent).
> 
> Currently I do this but have have to change the visibility after it is published. I would love to avoid having it visible for a split second and having the cache updated twice.

This is not in Create/Update Struct API because the entire operation is costly. We can consider adding it, but it's out of scope now.

Reply:
@wizhippo I think that it should be handled as is as `isHidden` is a part of `$location->contentInfo` (ref: https://github.com/ezsystems/ezplatform-admin-ui/blob/1.5/src/bundle/Controller/LocationController.php#L191) but isn't propagated on publishing content in the new location.

Reply:
> Is it not location that handles visibility? Should not locations be updated?

@wizhippo Content can be hidden on two different levels:
 - Location level: if you want single, particular location to be hidden,
 - Content level: if you want to hide content as a whole without loosing information about actual visibility of its locations (as you wouldn't be able to reveal content without keeping trace which locations have to be revealed and which were hidden intentionally and should remain hidden).

We need to keep the value of `isHidden` content flag during content operations.

Reply:
@Nattfarinn Thanks, forgot about the content level.